libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -255,7 +257,19 @@ void AP_OpenDroneID::send_location_message()
     if (!ahrs.get_location(current_location)) {
```

Comment:
As @peterbarker has mentioned, if this `get_location` returns false, it seems that `current_location` is still populated due to the dead-reckoning system. I agree with him that it would still make sense to publish this location, but set the status to `MAV_ODID_STATUS_REMOTE_ID_SYSTEM_FAILURE`, which leaves the possibility of recovering once we can get the current location.

Reply:
i've added an option for non-GPS operation, and declare an emergency for no-GPS by default